osconfig_tests/gcp_clients/gcp_clients.go

```diff
@@ -0,0 +1,80 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package gcpclients
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/config"
+	"google.golang.org/api/option"
+)
+
+var (
+	storageClient  *storage.Client
+	osconfigClient *osconfig.Client
+
+	populateClientOnce sync.Once
+)
+
+func PopulateClients(ctx context.Context) {
+	populateClientOnce.Do(func() {
```

Comment:
Nice! How about calling this function from `create*Client()` instead of forcing consumers to call this first?

Reply:
> Nice! How about calling this function from `create*Client()` instead of forcing consumers to call this first?
you mean `Get*Client()` ? 
when i relate this to OOP, this should be a part of object initialization which essentially mean dependency injection.
Dependency injection in functional programming language  has been bothering me for a while, now. You are right, consumer should not have to call a method to initialize dependency objects.
I feel `init() ` is Go idiomatic way to perform such tasks. but having faced the issue with initialization sequence of "init methods`, i am reluctant to use it anymore.

calling the populate method from `Get*Client()` is ok but then the initialization of the clients might not be consistent since initializtion would depend on which Get*client method was invoked first. This might be a bit annoying while debugging an issue.
if you agree, i would prefer to solve this the correct way which is to use DI, in a different PR

Reply:
SGTM. Addressing `SetConfig` would be good too. I've been bit by that before :snake:.